src/toggle-switch/toggle-switch.scss

```diff
@@ -0,0 +1,235 @@
+
+    // This is the most straightforward way of setting the knob's styles when the
+    // switch is both checked and disabled.
+
+    // stylelint-disable-next-line selector-max-specificity
+    .ToggleSwitch-knob {
+      background-color: var(--color-switch-knob-checked-disabled-bg);
+    }
+  }
+
+  .ToggleSwitch-knob {
+    background-color: var(--color-switch-knob-checked-disabled-bg);
+  }
+}
+
+.ToggleSwitch-track[aria-checked='true'] {
+  background-color: var(--color-switch-track-checked-bg);
+  border-color: var(--color-switch-track-checked-border);
+
+  &:hover {
+    background-color: var(--color-switch-track-checked-hover-bg);
+  }
+
+  &:active {
+    background-color: var(--color-switch-track-checked-active-bg);
+  }
+
+  .ToggleSwitch-knob {
+    background-color: var(--color-switch-knob-checked-bg);
+    border: 0;
+    transform: translateX(calc(100% + 1px));
+  }
+
+  .ToggleSwitch-lineIcon {
+    transform: translateX(0%);
+  }
+
+  .ToggleSwitch-circleIcon {
+    transform: translateX(100%);
+  }
+}
+
+.ToggleSwitch-track[aria-disabled='true'] {
+  cursor: not-allowed;
+  background-color: var(--color-canvas-subtle);
+  border-color: var(--color-border-subtle);
+  transition-property: none;
+
+  &:hover,
+  &:active {
+    .ToggleSwitch-knob {
+      background-color: var(--color-btn-bg);
+    }
+  }
+
+  .ToggleSwitch-knob {
+    border-color: var(--color-border-default);
+    box-shadow: none;
+
+    &:hover,
+    &:active {
+      background-color: var(--color-btn-bg);
+    }
+  }
+
+  .ToggleSwitch-lineIcon {
+    color: var(--color-fg-subtle);
+  }
+
+  .ToggleSwitch-circleIcon {
+    color: var(--color-fg-subtle);
+  }
+
+  .ToggleSwitch-labels {
+    color: var(--color-fg-muted);
+  }
+}
+
+.ToggleSwitch-bg {
+  display: flex;
+  align-items: center;
+  width: 100%;
+  height: 100%;
+  overflow: hidden;
+}
+
+.ToggleSwitch-lineIcon {
+  line-height: 0;
+  color: var(--color-accent-fg);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(-100%);
+  flex-shrink: 0;
+  flex-grow: 1;
+  flex-basis: 50%;
+}
+
+.ToggleSwitch-circleIcon {
+  line-height: 0;
+  color: var(--color-fg-default);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(0);
+  flex-shrink: 0;
+  flex-grow: 1;
+  flex-basis: 50%;
+}
+
+.ToggleSwitch-knob {
+  position: absolute;
+  top: -1px;
+  bottom: -1px;
+  z-index: 1;
+  width: 50%;
+  background-color: var(--color-btn-bg);
+  border-color: var(--color-switch-track-border);
+  border-style: $border-style;
+  border-width: $border-width;
+  border-radius: $border-radius;
+  box-shadow: var(--color-shadow-medium), var(--color-btn-inset-shadow);
+  transition-timing-function: cubic-bezier(0.5, 1, 0.89, 1);
+  transition-duration: 80ms;
+  transition-property: transform;
+  transform: translateX(-1px);
+
+  @media (prefers-reduced-motion) {
+    transition: none;
+  }
+}
+
+.ToggleSwitch-labels {
```

Comment:
Based on the [guideline](https://primer.style/design/components/toggle-switch#anatomy), should this be called `.ToggleSwitch-status`? And then maybe for the child elements:

- `ToggleSwitch-onLabel` -> `ToggleSwitch-statusOn`
- `ToggleSwitch-offLabel` -> `ToggleSwitch-statusOff`

Reply:
Oh nice yeah that makes sense 👍